typings/generated/src/gateway/types.d.ts

```diff
@@ -34,20 +34,20 @@ export interface NetworkGateway {
 }
 export interface XhrGateway extends Gateway, NetworkGateway {
     readonly withCredentials: boolean;
-    configure?: ((xmlHttpRequest: XMLHttpRequest) => void) | null;
-    configureBinary(xmlHttpRequest: XMLHttpRequest): void;
-    configureCallbacks(xmlHttpRequest: XMLHttpRequest): void;
-    configureTimeout(xmlHttpRequest: XMLHttpRequest): void;
-    createResponse(xmlHttpRequest: XMLHttpRequest): void;
-    createXHR(): XMLHttpRequest;
-    performRequest(method: string): void;
-    setHeaders(xmlHttpRequest: XMLHttpRequest, headers: Headers): void;
+    configure: ((xmlHttpRequest: XMLHttpRequest) => void) | null;
+    configureBinary?: (xmlHttpRequest: XMLHttpRequest) => void;
+    configureCallbacks?: (xmlHttpRequest: XMLHttpRequest) => void;
+    configureTimeout?: (xmlHttpRequest: XMLHttpRequest) => void;
+    createResponse?: (xmlHttpRequest: XMLHttpRequest) => void;
+    setHeaders?: (xmlHttpRequest: XMLHttpRequest, headers: Headers) => void;
+    createXHR?(): XMLHttpRequest;
+    performRequest?(method: string): void;
 }
 export interface GatewayConfiguration {
     Fetch: object;
     HTTP: HTTPGatewayConfiguration;
     Mock?: object;
     XHR: Partial<XhrGateway>;
-    enableHTTP408OnTimeouts?: boolean;
-    emulateHTTP?: boolean;
+    enableHTTP408OnTimeouts: boolean;
+    emulateHTTP: boolean;
```

Comment:
Are the changes here related/intended?

Reply:
Yes, most places reference it using `Partial<GatewayConfiguration>` though but this interface should be in line with what can be expected from the `gatewayConfigs` section of configuration (where it is set by default to `false`) so it is good to state that it will always be expected in mappersmith configs.

Reply:
Isn't this technically a breaking change? Are we envisioning that this PR will be a major change?

Reply:
References to the `GatewayConfiguration` interface: 
- `GatewayConstructor` (needs a `Partial<GatewayConfiguration>` - so it is unaffected)
- `GlobalConfigs` has a `gatewayConfigs` entry which is of type `GatewayConfiguration` meaning that even though it is internal as well, it is affected and the point of the change: `enableHTTP408OnTimeouts` and `emulateHTTP` are guaranteed to be present in `gatewayConfigs`.
- `ManifestOptions` (which represents the configuration object you pass to `forge`) has `gatewayConfigs?: Partial<GatewayConfiguration>` which means although this is the top level public interface to mappersmith, it's still unaffected by this change.